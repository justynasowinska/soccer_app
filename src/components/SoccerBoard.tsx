import * as React from 'react';
import { View, ImageBackground, StyleSheet } from 'react-native';
import { Player } from './Player';

import { data } from '../../assets/data';

export const SoccerBoard = () => {
    const playerPostions = data.player_positions[0];

    const countPosX = (pos: number) => {
        return `${pos * 100}%`;
    };

    const countPosY = (pos: number) => {
        return `${pos * 100}%`;
    }

    return (
        <View style={styles.soccerBoardContainer}>
            <ImageBackground
                source={require('../../assets/images/soccer.png')}
                resizeMode="contain"
                style={styles.imageBackground}
            >
                { playerPostions.map(player => {
                    return (
                        <Player
                            key={player[0]}
                            id={player[0]}
                            x={countPosX(player[1])}
                            y={countPosY(player[2])}
                        />
                    );
                }) }
            </ImageBackground>
        </View>
    );
};

const styles = StyleSheet.create({
    soccerBoardContainer: {
        width: '100%',
        flex: 1,
        alignItems: 'center',
    },
    imageBackground: {
        position: 'relative',
        height: '100%',
        aspectRatio: 2.2142286171,
    }
});