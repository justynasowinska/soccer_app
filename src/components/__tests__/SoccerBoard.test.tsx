import * as React from 'react';
import { shallow } from 'enzyme';

import { SoccerBoard } from '../SoccerBoard';

describe('SoccerBoard', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <SoccerBoard
                currentTime={1434}
            />
        );

        expect(component).toMatchSnapshot();
    });
});