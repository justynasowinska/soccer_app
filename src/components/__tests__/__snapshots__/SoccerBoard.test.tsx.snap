// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoccerBoard matches to the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "aspectRatio": 2.2142286171,
      "flex": 1,
      "justifyContent": "center",
      "maxWidth": "100%",
    }
  }
>
  <ImageBackground
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/images/soccer.png",
      }
    }
    style={
      Object {
        "aspectRatio": 2.2142286171,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Player
      id={13}
      key="13"
      x={0.211}
      y={0.257}
    />
    <Player
      id={2}
      key="2"
      x={0.263}
      y={0.422}
    />
    <Player
      id={5}
      key="5"
      x={0.336}
      y={0.45}
    />
    <Player
      id={1}
      key="1"
      x={0.505}
      y={0.642}
    />
    <Player
      id={15}
      key="15"
      x={0.479}
      y={0.367}
    />
    <Player
      id={14}
      key="14"
      x={0.266}
      y={0.266}
    />
    <Player
      id={7}
      key="7"
      x={0.399}
      y={0.559}
    />
    <Player
      id={8}
      key="8"
      x={0.272}
      y={0.579}
    />
    <Player
      id={9}
      key="9"
      x={0.307}
      y={0.187}
    />
    <Player
      id={10}
      key="10"
      x={0.43}
      y={0.213}
    />
    <Player
      id={12}
      key="12"
      x={0.71}
      y={1.045}
    />
  </ImageBackground>
</View>
`;
